inc/Extension/SyntaxPlugin.php

```diff
@@ -120,7 +121,7 @@ public function accepts($mode)
                 $this->allowedModes = array_merge($this->allowedModes, $PARSER_MODES[$mt]);
             }
 
-            $idx = array_search(substr(get_class($this), 7), (array)$this->allowedModes);
+            $idx = array_search(substr(get_class($this), 7), (array)$this->allowedModes, true);
```

Comment:
The default for the `$strict` parameter is [documented](https://www.php.net/manual/en/function.array-search) to be `false`. So this changes the logic. Is this intended?

Reply:
Ah I have disabled the strict checking rule in a later commit. This one must have slipped through earlier. It might be okay here though since only strings are searched here anyway.

Reply:
confirmed. strict checking should be fine here